core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -1168,18 +1168,28 @@ private void convertWhere(
     SqlNode newWhere = pushDownNotForIn(bb.scope(), where);
     replaceSubQueries(bb, newWhere, RelOptUtil.Logic.UNKNOWN_AS_FALSE);
     final RexNode convertedWhere = bb.convertExpression(newWhere);
-    final RexNode convertedWhere2 =
+    RexNode convertedWhere2 =
         RexUtil.removeNullabilityCast(typeFactory, convertedWhere);
 
+
+    //TODO: Support CSE in the where clause, https://bodo.atlassian.net/browse/BE-4092
+
     // only allocate filter if the condition is not TRUE
     if (convertedWhere2.isAlwaysTrue()) {
       return;
     }
 
     final RelFactories.FilterFactory filterFactory =
         RelFactories.DEFAULT_FILTER_FACTORY;
-    final RelNode filter =
-        filterFactory.createFilter(bb.root(), convertedWhere2, ImmutableSet.of());
+    final RelNode filter;
+    try {
+      filter =
+          filterFactory.createFilter(bb.root(), convertedWhere2, ImmutableSet.of());
+    } catch (Throwable e) {
+      System.out.println("TODO!");
```

Comment:
What is this TODO meant to handle? Does this need to be a proper exception?

Reply:
Sorry, this is left over debugging code, removing.